### PR TITLE
Add docs for sso-team-id to the Teams API

### PR DESCRIPTION
**Pull request by xlgmokha**
# Why?

The ability to manage team membership via SAML and the [MemberOf][member-of] attribute is
being expanded to allow mapping teams to a unique `sso-team-id`. This new
functionality is available to organizations with the SSO entitlement via the
[Business][business] tier.

# What?

This PR attempts to update the API documentation to show that a new field called
[`sso-team-id`][go-tfe] will appear in responses for teams that belong to an organization
with the appropriate entitlement. It also updates examples to show how to
create/update teams with an `sso-team-id`.

https://user-images.githubusercontent.com/80475/160010280-6e41337c-3283-476f-82f6-33d0d550d823.mov

Screen Shot 2022-03-24 at 15 38 53

[business]: https://www.hashicorp.com/products/terraform/pricing
[member-of]: https://www.terraform.io/enterprise/user-management/saml/attributes#memberof
[go-tfe]: https://github.com/hashicorp/go-tfe/pull/364

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201966179237923

### CODE REVIEW

File: content/cloud-docs/api-docs/teams.mdx
Comment (xlgmokha) [inline content/cloud-docs/api-docs/teams.mdx:90]: This field is only returned if the organization that this team belongs to is in the Business tier. How should I express this?

File: content/cloud-docs/api-docs/teams.mdx
Comment (xlgmokha) [inline content/cloud-docs/api-docs/teams.mdx:149]: This field is optional and is only usable if the team that is being created/updated belongs to an organization that is in the business tier.

**Comment by xlgmokha** [comment]
@laurapacilio would you be able to help me with some of the wording for this change. I'm having a bit of trouble expressing that this change is only available to organizations in the business tier.

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/teams.mdx:149]:
```suggestion
| `data.attributes.sso-team-id`           | string | (nothing)  | The unique identifier of the team from the SAML `MemberOf` attribute. Only available if the team belongs to an organization in the paid Terraform Cloud Business tier.                                                                                                                                                     |
```

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/teams.mdx:311]:
```suggestion
| `data.attributes.sso-team-id`           | string | (previous value) | The unique identifier of the team from the SAML `MemberOf` attribute.  Only available if the team belongs to an organization in the paid Terraform Cloud Business Tier.                                                                                                                                                   |
```

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [reply content/cloud-docs/api-docs/teams.mdx:90]: My instinct is to have one sentence under the Sample Response header that explains this.  Thoughts on this approach? I think it's better to state this in each place where the user could see this and potentially get confused. So each response that has this attribute in it would now have this brief explanatory line. I don't want to put it in a note because that feels like it would get visually very busy if we used it throughout the page. So something like...

### Sample Response

The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/teams.mdx:39]: What if we combined these two notes. Having two notes in a row like this can be visually overwhelming for users. Combining them into one note, especially since they are related (both paid features) I think is an improvement. Thoughts?

```suggestion
-> **Note:** Team management is a paid feature, available as part of the **Team** upgrade package. Single sign-on is also a paid feature, available as part of the **Business** upgrade package. Refer to [Terraform Cloud pricing](https://www.hashicorp.com/products/terraform/pricing) for details.
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
Some initial feedback for you! Let me know if you disagree or have other ideas :-) Otherwise, I'd love to see how this looks and take another editing pass once you've made the changes. Thank you for tagging me for review!

File: content/cloud-docs/api-docs/teams.mdx
Comment (JarrettSpiker) [reply content/cloud-docs/api-docs/teams.mdx:90]: Updated this based on your comment @laurapacilio

File: content/cloud-docs/api-docs/teams.mdx
Comment (JarrettSpiker) [inline content/cloud-docs/api-docs/teams.mdx:161]: Just wanted to call out that I **didn't** add a _The sso-team-id attribute is only returned if this team's organization is in the Terraform Cloud Business tier._ note here b/c it is called out in the table immediately above

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/teams.mdx:80]: I don't think we need italics!

```suggestion
The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.
```

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [reply content/cloud-docs/api-docs/teams.mdx:161]: okay that's fair and makes sense to me!

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/teams.mdx:183]:
```suggestion
The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.
```

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/teams.mdx:315]:
```suggestion
| `data.attributes.sso-team-id`           | string | (previous value) | The unique identifier of the team from the SAML `MemberOf` attribute. Only available if the team belongs to an organization in the paid Terraform Cloud Business tier.                                                                                                                                                   |
```

File: content/cloud-docs/api-docs/teams.mdx
Comment (laurapacilio) [inline content/cloud-docs/api-docs/teams.mdx:348]:
```suggestion
The `sso-team-id` attribute is only returned if this team's organization is in the Terraform Cloud Business tier.
```

**Review by laurapacilio — APPROVED** [review]
Approved with a few final nits :-) If you add those in, you have my support to merge! 🚀

File: content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx
Comment (JarrettSpiker) [inline content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx:72]: I thought to myself this morning that it might be nice to have a paragraph explaining _why_ SSO Team IDs are a thing, so I added a blurb here. I thought I could get it in before you re-reviewed @laurapacilio but no luck :(

Would you mind proofing this line to make sure you still approve?

Comment (laurapacilio) [inline content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx:72]:
```suggestion
The `MemberOf` SAML attribute may not be human readable. To maintain human readable team names in Terraform Cloud, you can set an SSO Team ID that matches the SAML attribute instead.
```

**Review by laurapacilio — APPROVED** [review]
thank you! a little correction for flow and then I approve :-)

File: content/cloud-docs/api-docs/teams.mdx
Comment (mpminardi) [inline content/cloud-docs/api-docs/teams.mdx:80]: I left a comment on [this](https://github.com/hashicorp/terraform-website/pull/2221) PR but will add it here as well. I imagine we'll want to mention here (and elsewhere that similar text has been added) that this is also going to exist in enterprise and potentially also add text that specifies what version the field will exist in there (e.g., `This attribute is also available in Terraform Enterprise as of version 202204-1` or something similar).

Comment (laurapacilio) [reply content/cloud-docs/api-docs/teams.mdx:80]: +1  But just to check - is the version we're listing already available? If not, I think we should wait until it is available before adding a future-facing statement.

"The `sso-team-id` attribute is only returned of this team's organization is in the Terraform Cloud Business tier or you set it in Terraform Enterprise version 202204-1 and later."

Because you have to have set it right? (I think that's what I read) I just don't want folks to think it'll always be there in TFE if it won't be.

Comment (JarrettSpiker) [reply content/cloud-docs/api-docs/teams.mdx:80]: > is the version we're listing already available? If not, I think we should wait until it is available before adding a future-facing statement.

It is not! We won't merge any docs changes (this one or #2221 ) until the changes are actually released.

> Because you have to have set it right? (I think that's what I read) I just don't want folks to think it'll always be there in TFE if it won't be.

You do need to set the value, yes. I will double check if the API returns the empty field (`sso-team-id : ""`) if you dont have it set, or if SSO isnt enabled...

**Comment by JarrettSpiker** [comment]
Unfortunately, when making the TFE docs side changes for this, I noticed that the exiting TFC docs (what this review modifies) are a bit misleading. Even before the SSO Team ID changes, they seem to imply that the SAML attribute which get's mapped to team names **has to be** `MemberOf`.

In fact, the TFE docs describe the actual behaviour, which is that it is configurable https://www.terraform.io/enterprise/user-management/saml/team-membership#configuring-team-membership-mapping

So I am going to update this review (again, sorry) to be closer aligned with the TFE docs

**Review by radditude — APPROVED** [review]

**Comment by JarrettSpiker** [comment]
> no worries - will you just let me know when its ready again?

@laurapacilio Updated! I mostly just pulled text from [the TFE](https://www.terraform.io/enterprise/user-management/saml/team-membership#configuring-team-membership-mapping) page, and tweaked it to be Cloud specific

**Comment by xlgmokha** [comment]
@laurapacilio, @JarrettSpiker do you think it's safe to merge this change?

File: content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx
Comment (laurapacilio) [inline content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx:83]:
```suggestion
Terraform Cloud expects the team names in the team membership SAML attribute to exactly match its own team names or its configured SSO Team IDs. This match is case sensitive.
```

**Review by laurapacilio — APPROVED** [review]
A couple of suggested nits for flow :) A lot of it is changing things to be active voice, but please push back on me if I've materially changed the meaning!

File: content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx
Comment (JarrettSpiker) [reply content/cloud-docs/users-teams-organizations/single-sign-on/index.mdx]: Sure! Updated

**Pull request merged**
